src/frontend/hooks/useProjectInviteListener.ts

```diff
@@ -1,14 +1,25 @@
-import * as React from "react";
-import { useNavigation } from "react-navigation-hooks";
-import ProjectInviteContext from "../context/ProjectInviteContext";
-
-export default function useProjectInviteListener() {
-  const navigation = useNavigation();
-  const { invite } = React.useContext(ProjectInviteContext);
-
-  React.useEffect(() => {
-    if (invite) {
-      navigation.navigate("ProjectInviteModal");
-    }
-  }, [invite, navigation]);
+import { useEffect, useRef } from "react";
+
+// Gregor TODO: Add method to API for (un)subscribing to project invites
+const subscribeToProjectInvites = (callback: (inviteKey: string) => void) => {
+  const intervalId = setInterval(() => callback("abc123"), 3000);
+  return () => clearInterval(intervalId);
+};
+
+export default function useProjectInviteListener(
+  onInviteReceived: (inviteKey: string) => void
+) {
+  const callbackRef = useRef(onInviteReceived);
+
+  useEffect(() => {
+    callbackRef.current = onInviteReceived;
+  }, [onInviteReceived]);
```

Comment:
Im a little confused, where are we calling the function, onInviteReceived(), here?

Reply:
we're not calling it here, we're just updating the ref value whenever `onInviteReceived` does change. This is so that we don't potentially cause the effect that handles the subscription logic multiple times, but when the subscription is triggered, it uses an updated value of the callback that we provided.

you'll notice that in the useEffect with the subscription, we call `callbackRef.current`, which refers to the `onInviteRecieved` callback that's provided

Reply:
the `useEventListener` hook [here](https://usehooks.com/useEventListener/) has similar implementation that does a good job explaining this pattern and why it's useful:

```js

  // Update ref.current value if handler changes.
  // This allows our effect below to always get latest handler ...
  // ... without us needing to pass it in effect deps array ...
  // ... and potentially cause effect to re-run every render.
  useEffect(() => {
    savedHandler.current = handler;
  }, [handler]);
```